docs/miscellaneous.rst

```diff
@@ -88,6 +92,264 @@ by checking if the lowest bit is set: short (not set) and long (set).
 .. note::
   Handling invalidly encoded slots is currently not supported but may be added in the future.
 
+JSON Output
+===========
+
+.. _storage-layout-top-level:
+
+The storage layout of a contract can be requested via the :ref:`standard JSON interface <compiler-api>`.  The output is a JSON object containing two keys,
+``storage`` and ``types``.  The ``storage`` object is an array where each
+element has the following form:
+
+
+.. code::
+
+
+    {
+        "astId":"2",
+        "contract":"A",
+        "label":"x",
+        "offset":"0",
+        "slot":"0",
```

Comment:
Here it might make sense to use a string because json numbers are probably not large enough.

Reply:
Yea, that's why I went with string for all

Reply:
Maybe mention somewhere that it is a string? Or is there no real place?

Reply:
And also mention that it can be very large.

Reply:
Mentioned it in the first part of the doc